enhancements/oc/olm-mirroring.md

```diff
@@ -0,0 +1,121 @@
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+This enhancement outlines the changes necessary to `oc` to support disconnected distribution of OLM catalogs. It relies on the [Disconnected OLM enhancement](https://github.com/operator-framework/enhancements/blob/master/enhancements/opm-mirroring.md).
+
+## Motivation
+
+Disconnected installation of OLM catalogs can be done with the following `opm` commands:
+
+```
+$ opm registry build build --appregistry-namespace=quay.io/community-operators --to disconnected-registry:5000/openshift/community-operators-catalog:4.2.1 
+pulling quay.io/community-operators/etcd...
+buliding catalog...
+mirroring catalog...
+quay.io/community-operators mirrored to disconnected-registry:5000/openshift/community-operators-catalog:4.2.1
+
+$ docker pull disconnected-registry:5000/openshift/community-operators-catalog:4.2.1
+
+$ opm registry images --from=disconnected-registry:5000/openshift/community-operators-catalog:4.2.1 --to=disconnected-registry:5000/community-operators --manifests=./mirror-manifests | xargs oc image mirror
+
+$ oc apply -f ./mirror-manifests
+```
+
+This requires both `oc` and `opm` to be installed and available on bastion hosts. We would like to distribute one tool for creating disconnected tools, not two.
+
+### Goals
+* Define `oc` commands which can build and mirror all images required for mirroring to a disconnected envorinment
+
+### Non-Goals
+* Provide an api for fine-grained control over which operators / operands are mirrored. 
+
+## Proposal
+
+The basic approach will be to:
+
+- Generate catalog container images that can replace the appregistry repositories
+- For each operator within the catalog image, mirror it to the disconnected environment
+
+### Generate and mirror a catalog
+
+`oc` will be extended with a command (equivalent to the `opm` command)
+
+```sh
+$ oc adm catalog build --appregistry-namespace=community-operators --to quay.io/ecordell/community-operators-catalog:4.2.1 
+```
+
+`oc adm catalog build` will:
+
+- Use appregistry protocol to retrieve all of the operators in a namespace of Quay.io specified by `appregistry-namespace`
+- Load all of the downloaded operators into a versioned `operator-registry` sqlite artifact
+- Build a runnable operator-registry image by appending the database to the `operator-registry` base image using the same machinary as `oc image append`
+- Mirror that image to the target defined by `--to`, using the same machinary as `oc image append`
+
+It will have the following flags:
+
+- `--from=ref` - the base image to add the built operator database into. Defaults to the operator-registry image shipped with the version of OpenShift that `oc` came from.
+- `--to=ref` - the location that the image will be mirrored.
+- `--save=path` - instead of mirroring, this will output a tarball containing the image
```

Comment:
`--dir` will be better, especially that you need to combine this with `--db-only`